src/main/java/com/ays/assignment/controller/AssignmentController.java

```diff
@@ -232,4 +221,19 @@ public AysResponse<Void> completeAssignment() {
         return AysResponse.SUCCESS;
     }
 
+    /**
+     * Cancels assignment that is either started by user or assigned to user.
+     * Requires USER authority.
+     *
+     * @param cancelRequest The request object containing cancel reason.
+     * @return A success response if assignment canceled.
+     */
+    @PostMapping("/assignment/cancel")
+    @PreAuthorize("hasAnyAuthority('USER')")
+    @SuppressWarnings("unused")
```

Comment:
Bu annotation gerekli mi?

Reply:
Reason bilgisini alıyoruz fakat henüz için bir loglama sistemi olmadığı için kullanmıyoruz.
Dolayısıyla editör bu parametreyi kullanmadın hatası göstermesin şimdilik istiyorum.

Reply:
Burası için almamalıyız aslında, Servise request gönderilmediği için burada unused alıyoruz, o objeyi göndermeliyiz, kullanıp kullanmaması servisin elinde, loglamayı da servis katmanında yapabiliriz

Reply:
İstek sınıfını servise taşıdım. Dolayısıyla bu anotasyona burda gerek kalmadı.